### DISCUSSION TITLE
Will rearranging or renaming files from my directory outside of Stash make files disappear in Stash?

### DISCUSSION THREAD

**Discussion by user50237** [stashapp/stash#3309] [Q&A] [answered]
I want to know if I can rearrange or rename my files using File Explorer without it disappearing from Stash or affecting the file's tags, rating etc.

**Comment by stg-annon** [comment, marked as answer]
they will not disappear from stash unless you perform a clean operation, for stash to rediscover the files you will need to re-scan them and stash will automatically pick them back up, as all files are ID'd via hash
